build_dev_dockers.yml

```diff
@@ -0,0 +1,108 @@
+
+# Usage:
+#      $ cp vars/settings.yml.template vars/settings.yml
+#
+#      Fill out vars/settings.yml.
+#
+#      $ ansible-playbook -i TARGET-HOST build_dockers.yml [OPTIONS]...
+#
+# Required parameters should be specified in vars/settings.yml or --extra-vars
+# ansible command line parameter. For details, see vars/settings.yml.template.
+#
+# Examples:
+# To run on your localhost (Note comma is needed after localhost):
+# $ ansible-playbook -i localhost, build_dockers.yml -c local
+#
+# To run this on a remote host, say build01, as root:
+# $ ansible-playbook -i build01, build_dockers.yml  \
+#       --become --become-user root --ask-become-pass
+- hosts: all
+  vars_files:
+    - vars/settings.yml
+  tasks:
+    - assert:
+        that: ansible_version.major >= 2
+        msg: "Only supports ansible version 2.x."
+
+    - assert:
+        that: spark_kubernetes_jar_path is defined
+        msg: >
+          'spark_kubernetes_jar_path' should point to a path of a Spark kubernetes
+          jar.  See vars/settings.yml.template for details.
+
+    - assert:
+        that: docker_registry is defined
+        msg: >
+          'docker_registry' should specify docker registry host and port.
+          See vars/settings.yml.template for details.
+
+    - assert:
+        that: docker_base_image is defined
+        msg: >
+          'docker_base_image' should specify the base image to be used to create docker images.
+          See vars/settings.yml.template for details.
+
+    - assert:
+        that: docker_tag is defined
+        msg: >
+          'docker_tag' should specify docker tag string.
+          See vars/settings.yml.template for details.
+
+    - name: Create a temporary workdir
+      command: mktemp -d /tmp/build-images-workdir-XXXXX
+      register: _workdir_register
+
+    - name: Set workdir as fact
+      set_fact: _workdir="{{ _workdir_register.stdout }}"
+
+    - name: Creates the jar directory in workspace
+      file: path="{{ _workdir }}/jars" state=directory
+
+    - name: Copy the jar file to workspace jars directory
+      copy: src="{{ spark_kubernetes_jar_path }}" dest="{{ _workdir }}/jars" remote_src=True
+
+    - name : Generate Dockerfile content
+      set_fact: dockerfile_content="FROM {{ docker_base_image }}\nCOPY jars /opt/spark/jars\n"
+
+    # Create a blank file
+    - name: Create an empty Dockerfile
+      file: path="{{ _workdir }}/Dockerfile" state=touch
+
+    - name: Write to Dockerfile
+      copy: content="{{ dockerfile_content }}" dest="{{ _workdir }}/Dockerfile"
+
+    - block:
+        # NOTE. The unarchive module is broken with macOS BSD tar. Install
+        # GNU tar and make sure your path points it for the tar command.
+        # See https://github.com/ansible/ansible-modules-core/issues/3952.
+        - name: Build and push images
+          shell: >
+            docker build  \
+                -t {{ docker_registry }}/spark-{{ item }}:{{ docker_tag }}  \
```

Comment:
s/docker_tag/dev_docker_tag/ so that settings file can specify tags for base image and child image separately.